pkgs/development/haskell-modules/default.nix

```diff
@@ -36,6 +37,8 @@ let
     compilerConfig
     packageSetConfig
     overrides
+  ] ++ lib.optionals stdenv.hostPlatform.isGhcjs [
+    (configurationJS { inherit pkgs haskellLib; })
```

Comment:
I'd prefer if we could just use `platformConfigurations` for this instead.

Reply:
Agreed. This now works, as per: https://github.com/NixOS/nixpkgs/pull/309650#discussion_r1640253924